geanylua/glspi_doc.c

```diff
@@ -92,34 +83,51 @@ static const gchar* doc_idx_to_filename(gint idx) {
 
 
 
-/* Actvate and focus the specified document */
+/* Switch to the specified document tab */
 static gint glspi_activate(lua_State* L)
 {
-	gint idx=-1;
-	if (lua_gettop(L)>0) {
-		if (lua_isnumber(L,1)) {
-			idx=(lua_tonumber(L,1));
-			if (idx<0) { /* Negative number refers to (absolute) GtkNotebook index */
-				idx=(0-idx)-1;
-				if (idx>=gtk_notebook_get_n_pages(NOTEBOOK)) { idx=-1;}
-			} else { /* A positive number refers to the geany->documents_array index */
-				idx=doc_idx_to_tab_idx(idx-1);
+	gint lua_idx = -1;
+	gint tab_idx = -1;
+
+	if (lua_gettop(L) > 0) {
+		if (lua_isnumber(L, 1)) {
```

Comment:
shouldn't that be [`lua_isinteger()`](https://www.lua.org/manual/5.4/manual.html#lua_isinteger) rather than [`lua_isnumber()`](https://www.lua.org/manual/5.4/manual.html#lua_isnumber) as it's converted using [`lua_tointeger()`](https://www.lua.org/manual/5.4/manual.html#lua_tointeger) in the body?  Note that I don't know the details, but there seem to be more difference than float vs. integer.

Reply:
OK. Did you check how it behaves if called with a string representing a number or a real number rather than an integer?
It's probably harmless but that might change behavior of this function with such values.

Reply:
I tested with following:
```
test = geany.activate("2.6")
if (test)
then
  geany.status("geany.activate success")
else
  geany.status("geany.activate fail")
end
```
It truncates to `2`.

Reply:
Which Lua version? But anyway using a string here seems horrible anyway so that's just for compatibility

Reply:
BTW, unrelated to this, but it seems an awful idea to allow a string representation of a number, and then a string for the file name: this means that if the filename is e.g. `"1.2"` it's not gonna work.
Swapping the integer/string checks would be slightly incompatible but probably a future change worth considering.

Reply:
Right now, geanylua supports only 5.1.  I plan to look at that after another bugfix #1397 is merged.

Reply:
From comment in code and brief test, string has to be full document path to work.

Reply:
Ah, OK, fair enough